### PR TITLE
add changelog for 1.2.0-rc.1

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Add Changelog for `1.2.0-rc.1` in `CHANGELOG.md

### CODE REVIEW

**Review by JunDai — APPROVED** [review]
🚢

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,20 @@
...
 
+## Mapbox Navigation SDK 1.2.0-rc.1 - November 11, 2020
+Visit our [1.0.0 migration guide](https://github.com/mapbox/mapbox-navigation-android/wiki/1.0-Navigation-SDK-Migration-Guide) and the [documentation pages](http://docs.mapbox.com/android/navigation/overview) for more information.
+
+### Changelog
+#### Improvements
+* Remove electronic horizon experimental note. [#3748](https://github.com/mapbox/mapbox-navigation-android/pull/3748)
+* Upload Free Drive telemetry events to the server. [#3751](https://github.com/mapbox/mapbox-navigation-android/pull/3751)
+* Track appropriate telemetry events for waypoints. [#3747](https://github.com/mapbox/mapbox-navigation-android/pull/3747)
+* Introduce `NavigationMapboxMap.Builder` and expose vanishing point. [#3745](https://github.com/mapbox/mapbox-navigation-android/pull/3745)
```
Comment (LukasPaczos) [inline CHANGELOG.md:13]:
```suggestion
* Introduce `NavigationMapboxMap.Builder` and expose vanishing point update interval setting. [#3745](https://github.com/mapbox/mapbox-navigation-android/pull/3745)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,20 @@
...
+
+#### Bug fixes
+
+* Include the last segment of `RouteAlertGeometry` represented in the `LineString` generated by `RouteAlertUtils` [#3744](https://github.com/mapbox/mapbox-navigation-android/pull/3744)
```
Comment (LukasPaczos) [inline CHANGELOG.md:17]:
```suggestion
* Include the last segment of `RouteAlertGeometry` represented in the `LineString` generated by `RouteAlertUtils`. [#3744](https://github.com/mapbox/mapbox-navigation-android/pull/3744)
```

**Review by LukasPaczos — APPROVED** [review]
For changes like these feel free to add the `skip changelog` label.

**Comment by LukasPaczos** [comment]
It'd also be good to add changelog entries from https://github.com/mapbox/mapbox-navigation-android/pull/3761.

**Review by Guardiola31337 — COMMENTED** [review]
Minor thing. We've been using the _past tense_ for the `CHANGELOG` but not this time, should we update?

**Review by Guardiola31337 — APPROVED** [review]

**Pull request merged**